contrib/pkg/awstagdeprovision/awstagdeprovision.go

```diff
@@ -354,6 +455,12 @@ func deleteVPCs(awsSession *session.Session, filters AWSFilter, clusterName stri
 				return false, nil
 			}
 
+			complete = deleteV2LBs(vpc, awsSession, logger)
+			if !complete {
+				logger.Debugf("not finished deleting V2 load balancers, will need to retry")
```

Comment:
This whole block seems inefficient to me.  Can we try all of the children (`deleteLBs`, `deleteV2LBs`, and `deleteRouteTablesWithVPC`) even if one of them errors or is unfinished?  You could do that with something like:

```go
v1LBComplete := deleteLBs(vpc, awsSession, logger)
v2LBComplete := deleteV2LBs(vpc, awsSession, logger)
err := deleteRouteTablesWithVPC(vpc, ec2Client, logger)
if err != nil {
  logger.Debugf("error deleting route tables: %v", err)
  return false, nil
}
if !v1LBComplete {
  logger.Debugf("not finished deleting load balancers, will need to retry")
  return false, nil
}
if !v2LBComplete {
  logger.Debugf("not finished deleting V2 load balancers, will need to retry")
  return false, nil
}
```

That's still serial, but at least we try them all.  You could use goroutines or some such to make those parallel.  I'm also fine punting either my snippet above or parallel calls to follow-up work.

Reply:
deleteRouteTablesWithVPC will not succeed unless all the consumers of the route tables are it might be waste of api calls.

Reply:
> `deleteRouteTablesWithVPC` will not succeed unless all the consumers of the route tables are deleted; So it might have been setup to prevent waste of api calls.

Ok.  And we don't expect any clusters with both v1 and v2 load balancers in the same VPC.  So I'm fine with this as you have it, then :).

Reply:
If I spun up a cluster with these new LBs and then used Service type LoadBalancer, would I end up with V1 LBs as well?

Reply:
>  And we don't expect any clusters with both v1 and v2 load balancers in the same VPC
we do expect that there be both v1 and v2 LBs in the same VPC

@wking would there be any problems in that case?

Reply:
It looks like it would work fine to me.

Reply:
> we do expect that there be both v1 and v2 LBs in the same VPC.

Oh, in that case I'm back in favor of something like [this][1] to get at least v1 and v2 LB deletion attempts (but not route table deletions?) in the same round of attempted-VPC-deletion.  Otherwise we' could go through a few rounds trying to delete the v1 LBs, and then another few rounds trying to delete the v2 LBs, and end up deeper in the exponential backoff making teardown slower than it needs to be.

[1]: https://github.com/openshift/hive/pull/62#discussion_r230140110

Reply:
@wking with `https://github.com/openshift/hive/pull/62/commits/b4602eda58f2498f1925ef18d020bc4f98a80782` to `https://github.com/openshift/hive/pull/62/commits/004e9e45c2affde0176c3362ac8ffd6c9f348dd0` trying to delete both v1 and v2 and bailing if any one fails/not complete.